app/main.py

```diff
@@ -1,3 +1,49 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
+        data = json.load(f)
+        fuel_price = data["FUEL_PRICE"]
+        for customer_data in data["customers"]:
+            customer = Customer(**customer_data)
+            print(f"{customer.name} has {customer.money} dollars")
+            closest = {}
+            shop_names = {}
+            for shop in Shop.create_shops("app/config.json"):
+                fuel = customer.car["fuel_consumption"]
+                expenses = (
+                    customer.product_total(shop.products)
+                    + customer.fuel_expenses(fuel, shop.location, fuel_price)
+                )
+                closest[shop.name] = expenses
+                shop_names[shop.name] = shop.products
+                print(
+                    f"{customer.name}'s trip "
+                    f"to the {shop.name} costs {expenses}"
+                )
+            if customer.money > expenses:
+                closest_shop_name = min(closest, key=closest.get)
+                print(f"{customer.name} rides to {closest_shop_name}")
+            else:
+                print(
+                    f"{customer.name} doesn't have "
+                    f"enough money to make a purchase in any shop"
+                )
+                break
```

Comment:
break is stop cycle for customers
you need to use continue to skip current element